### PR TITLE
Parametrize number of similar questions generated

### PR DESCRIPTION
**Pull request by daureng**
# Summary
Parametrized the number of similar questions generated by llm. If it is set to 0, similar questions are not generated.

# Tests
Tested manually using different numbers as parameters

### CODE REVIEW

File: gen_ai/llm.yaml [hunk 1]
```diff
@@ -204,7 +205,7 @@ similar_questions_prompt: |
     - If the question is very short, anticipate what else they may want to know.
     - Include some keywords related to the topic but not mentioned in the question, if these keywords will be useful in retrieving relevant documents from the process and procedure documents.
     - Return only the similar questions without any introductory text.
-    - Limit the number of similar questions to 2.
+    - Limit the number of similar questions to {similar_questions_number}.
```
Comment (misha-chertushkin) [inline gen_ai/llm.yaml:208]: I guess you need to pass the similar_questions_number as the parameter to the LLMChain, so you need to modify provide_chain in ioc_container.py

File: gen_ai/common/retriever.py [hunk 2]
```diff
@@ -130,17 +130,24 @@ def retrieve_initial_documents(
...
 
     if "?" not in question:
         question = f"{question}?"
-    similar_questions = similar_questions_chain.run(question)
-    Container.logger().info(msg=f"Questions:\n {similar_questions}")
+        
+    similar_questions = ""
+    similar_questions_number = Container.config["similar_questions_number"]
+
+    if similar_questions_number:
```
Comment (misha-chertushkin) [inline gen_ai/common/retriever.py:140]: if similar questions is 0 lets just populate questions = [question] and completely skip any other logic

**Review by misha-chertushkin — CHANGES_REQUESTED** [review]
Looks good, posted 2 small suggestions

File: gen_ai/common/retriever.py (hunk 2 above)
Comment (daureng) [reply gen_ai/common/retriever.py:140]: good one, fixed

File: gen_ai/llm.yaml (hunk 1 above)
Comment (daureng) [reply gen_ai/llm.yaml:208]: Fixed, how did it even work

**Review by misha-chertushkin — APPROVED** [review]
Great job! LGTM

**Review by anna-novakovska — APPROVED** [review]
lgtm

**Pull request merged**